apollo-router/src/query_planner/fetch.rs

```diff
@@ -127,6 +128,9 @@ pub(crate) struct FetchNode {
     // authorization metadata for the subgraph query
     #[serde(default)]
     pub(crate) authorization: Arc<CacheKeyMetadata>,
+
+    #[serde(skip)]
+    pub(crate) executable_document: Arc<apollo_compiler::ExecutableDocument>,
```

Comment:
I believe this `serde(skip)` means that deserializing a query plan from distributed Redis cache will produce an empty document.

I could maybe be fixed by adding calls to `parse_subgraph_operations` in more places, but I think a better design is to have the fetch node itself lazily initialize the document on demand.

I’ll make a pull request that targets this branch to show what I have in mind.